### ISSUE TITLE
Tile function returns incorrect values

### ISSUE THREAD

**Issue by tlikhomanenko** [arrayfire/arrayfire#2348]
Hey!

There is an example when **tile** function returns incorrect values:

```
#include <arrayfire.h>
#include <array>
#include <iostream>

using namespace af;
using namespace std;

// Error in `tile` output

int main() {
  af::info();
  int batch = 11; // works for 12, 14, 16, 18, 21; doesn't work for 11, 13, 15, 17, 19, 20
  int channels = 221452;
  int dim = 0;
  af::array d_input = constant(3, channels, 5, batch);
  auto maxvals = max(d_input, dim);

  // Tile method 1
  dim4 tiledims(1, 1, 1, 1);
  tiledims[dim] = d_input.dims(dim);
  auto tile_result = tile(maxvals, tiledims);
  std::cout << "Tile output:  min " << af::min<float>(tile_result) << " max "
            << af::max<float>(tile_result) << "\n";

  return 0;
}
```

with the output (min and max values should be 3):
```
ArrayFire v3.6.1 (CUDA, 64-bit Linux, build default)
Platform: CUDA Toolkit 9.2, Driver: 396.26
[0] Tesla M40, 11449 MB, CUDA Compute 5.2
-1- Tesla M40, 11449 MB, CUDA Compute 5.2
Tile output:  min 0 max 3
```

**Comment by mlloreda** [comment]
Hey @tlikhomanenko,

Thanks for taking the time to report this! It appears the issue has been fixed in master (and the v3.6 branch, which we are working on releasing soon). In the meantime, you may build from source in order to use latest code.

**Comment by tlikhomanenko** [comment]
It seems that this fix works fine. Thanks!

**Comment by andresy** [comment]
fyi this has been fixed with 141ef4589b2f3a1454918da1dd8cb2a695535f30 which is included in 3.6.2 (but not 3.6.1).
